src/sagemaker_algorithm_toolkit/hyperparameter_validation.py

```diff
@@ -176,6 +177,50 @@ def validate_range(self, value):
             raise exc.UserError("Hyperparameter {}: value {} not in range {}".format(self.name, value, self.range))
 
 
+class NestedListHyperparameter(Hyperparameter):
+    def __init__(self, *args, **kwargs):
+        if kwargs.get("range") is None:
+            raise exc.AlgorithmError("range must be specified")
+        super(NestedListHyperparameter, self).__init__(*args, **kwargs)
+
+    def parse(self, value):
+        if isinstance(value, str):
+            return ast.literal_eval(value)
+        elif isinstance(value, list):
+            return value
+
+    def format_range(self):
+        min_, max_ = self.range.format_as_integer()
+        return {"NestedParameterRangeSpecification": {
+            "MinValue": min_,
+            "MaxValue": max_}}
+
+    def validate_range(self, value):
+        if any([index not in self.range for outer in value for index in outer]):
+            raise exc.UserError("Hyperparameter {}: value {} not in range {}".format(self.name, value, self.range))
+
+
+class TupleHyperparameter(Hyperparameter):
+    def __init__(self, *args, **kwargs):
+        if kwargs.get("range") is None:
+            raise exc.AlgorithmError("range must be specified")
+        super(TupleHyperparameter, self).__init__(*args, **kwargs)
+
+    def parse(self, value):
+        if isinstance(value, str):
+            return eval(value)
+        elif isinstance(value, list) or isinstance(value, tuple):
```

Comment:
I'm not convinced on the correctness of the design here. Shouldn't we fail validation if the value is a list?

Reply:
The [documentation](https://xgboost.readthedocs.io/en/latest/tutorials/monotonic.html) about montone constraints is not clear regarding the accepted format.
This was [where](https://github.com/dmlc/xgboost/issues/1514) I found out that python supports tuple format but this issue was raised in 2016 so added 'list' to not fail a validation.
I modified it to support only 'tuple' in the new commit.